content/posts/2019-08-03---Going-from-ember-to-vue.md

```diff
@@ -71,13 +75,19 @@ The JavaScript of a component is too tightly tied to the markup (as it should be
 ## State management
 
 The popular state management solution seems to be [Vuex](https://vuex.vuejs.org/) (another "official" Vue library written by core team members).
-This is much closer to [Redux](https://redux.js.org/) or [NGRX](https://ngrx.io/).
-I'm not going to claim that these patterns are the end-all-be-all solutions for state management on the front-end, but it's where most of the community seems to be going right now.
-Which is a huge plus over Ember.
-The skills/practices you pick up writing Vuex will translate to these other libraries.
+This is much closer to [Redux](https://redux.js.org/) or [NGRX](https://ngrx.io/) -- if you're familiar with those libraries, you'll be able to jump right into Vuex.
+I'm not going to claim that these libraries are the end-all-be-all solutions for state management on the front-end, but it's where most of the community seems to be going right now.
+Which I feel is a huge plus over Ember, since the skills/practices you pick up writing Vuex will translate over to React/Redux or something similar.
 
 You are also much more explicit when updating your "store" and that makes it feel so much less magical than Ember Data.
-No, there's no identity mapping or convenient CRUD methods, but that also means you get to be more flexible with how you request data (and as a result, you'll probably be much more aware of what is going on in your application).
+In Ember data, when I call `post.save()`, it isn't immediately clear what is happening (especially as a new developer).
+It makes a `POST` request if the record is new, a `PUT` request if it is not, updates the record in our store, looks for the id being returned from the API for updates, updates our store record, etc.
+Which isn't to say that stuff is bad, but when I was first getting into Ember, I found it confusing (same with `findAll` versus `peekAll`).
```

Comment:
if these ore going to be mentioned, it may be beneficial to say that findAll makes a network request and peekAll only looks at cache, which is super handy for avoiding prop-drilling

Reply:
If cache is already filled `findAll` is doing both of them at the same time (unless you specify `reload: true`).

Reply:
`findAll` will always make a request, it just checks the store first.